hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
```

Comment:
```[]``` sounds safer option to me. Or even raising a suitable exception (or is that too drastic?) Having a call to ```partition``` where ```n``` is 0 sounds like a bug to me and Hy probably should report that instead of trying to keep the system limping forward.